heat/core/rounding.py

```diff
@@ -328,6 +340,90 @@ def round(
 DNDarray.round.__doc__ = round.__doc__
 
 
+def sign(x: DNDarray, out: Optional[DNDarray] = None) -> DNDarray:
+    """
+    Returns an indication of the sign of a number, element-wise. The definition for complex values is equivalent to :math:`x / \\sqrt{x \\cdot x}`.
+
+    Parameters
+    ----------
+    x : DNDarray
+        Input array
+    out : DNDarray, optional
+        A location in which to store the results.
+
+    See Also
+    --------
+    :func:`sgn`
+        Equivalent function on non-complex arrays. The definition for complex values is equivalent to :math:`x / |x|`.
+
+    Examples
+    --------
+    >>> a = ht.array([-1, -0.5, 0, 0.5, 1])
+    >>> ht.sign(a)
+    DNDarray([-1., -1.,  0.,  1.,  1.], dtype=ht.float32, device=cpu:0, split=None)
+    >>> ht.sign(ht.array([5-2j, 3+4j]))
+    DNDarray([(1+0j), (1+0j)], dtype=ht.complex64, device=cpu:0, split=None)
+    """
+    # special case for complex values
+    if types.heat_type_is_complexfloating(x.dtype):
+        sanitation.sanitize_in(x)
+        if out is not None:
+            sanitation.sanitize_out(out, x.shape, x.split, x.device)
+            out.larray.copy_(x.larray)
+            data = out.larray
+        else:
+            data = torch.clone(x.larray)
+        # NOTE remove when min version >= 1.9
+        if "1.7" in torch.__version__ or "1.8" in torch.__version__:
+            pos = data != 0
+        else:  # pragma: no cover
+            indices = torch.nonzero(data)
+            pos = torch.split(indices, 1, 1)
```

Comment:
I'm missing something for sure so forgive the possibly stupid question, but why do we need to introduce a version dependency for such a simple operation?

The following runs fine on pytorch 1.9.:

```python
pos = data != 0
data[pos] = x.larray[pos] / torch.sqrt(torch.square(x.larray[pos]))
```

Reply:
There's a historical reason. I started writing the function with PyTorch 1.9. and I then added the special code for 1.7/1.8 because `nonzero` is not implemented for complex values there. At the same time, Heat pinned the required version of PyTorch and I've never came back to compare the code on 1.9. again :sweat_smile:

Reply:
On my local machine, the code for 1.9 is faster and uses less memory.